src/main/typescript/wcardinal/ui/d-table-header-cell.ts

```diff
@@ -307,4 +357,72 @@ export class DTableHeaderCell<
 
 		super.destroy();
 	}
+
+	protected onCellDown(e: InteractionEvent): void {
+		if (!this.isOnEdge(e) || this._isResizing) return;
+		const layer = DApplications.getLayer(this);
+		if (layer) {
+			this._isResizing = true;
+			const interactionManager = layer.renderer.plugins.interaction;
+			this._interactionManager = interactionManager;
+			const onCellResizeBound = this._onCellResizeBound;
+			interactionManager.on(UtilPointerEvent.move, onCellResizeBound);
+			const onCellUpBound = this._onCellUpBound;
+			interactionManager.on(UtilPointerEvent.up, onCellUpBound);
+			interactionManager.on(UtilPointerEvent.upoutside, onCellUpBound);
+			interactionManager.on(UtilPointerEvent.cancel, onCellUpBound);
+		}
+	}
+
+	protected onCellResize(e: InteractionEvent): void {
+		const point = new Point(0, 0);
+		this.toLocal(e.data.global, undefined, point);
+		const oldWidth = this.width;
+		const newWidth = Math.max(this.theme.getMinWidth(), point.x);
+		const ratio = newWidth / oldWidth;
+		this.width = newWidth;
+		let isWeight = false;
+		if (this.weight > 0) {
+			this.weight = this.weight * ratio;
+			isWeight = true;
+		}
+		this.emit("resize", newWidth, isWeight, this);
+	}
+
+	protected onCellUp(e: InteractionEvent): void {
+		if (!this._isResizing) return;
+		this._isResizing = false;
+		const interactionManager = this._interactionManager;
+		if (interactionManager) {
+			this._interactionManager = undefined;
+			const onCellResizeBound = this._onCellResizeBound;
+			interactionManager.off(UtilPointerEvent.move, onCellResizeBound);
+			const onCellUpBound = this._onCellUpBound;
+			interactionManager.off(UtilPointerEvent.up, onCellUpBound);
+			interactionManager.off(UtilPointerEvent.upoutside, onCellUpBound);
+			interactionManager.off(UtilPointerEvent.cancel, onCellUpBound);
+		}
+	}
+
+	protected isOnEdge(e: InteractionEvent): boolean {
+		const local = this.toLocal(e.data.global);
+		const localBounds = this.getLocalBounds();
+		const isInWidthRange = local.x >= 0 && local.x <= localBounds.width;
+		const isInHeightRange = local.y >= 0 && local.y <= localBounds.height;
+		if (isInHeightRange && isInWidthRange) {
+			const isOnEdge = localBounds.width - local.x < this.theme.getResizeWidth();
+			return isOnEdge;
+		}
+		return false;
+	}
+
+	protected onCellHover(e: InteractionEvent): void {
+		if (!this._isResizing) {
+			if (this.isOnEdge(e)) {
+				this.cursor = "col-resize";
```

Comment:
We are not allowed to assume cursor types here. The theme must be taken into the consideration.